source/Calamari.Shared/Integration/Scripting/PackagedScriptRunner.cs

```diff
@@ -45,25 +45,38 @@ protected void RunScripts(RunningDeployment deployment)
             }
         }
 
-        protected void DeleteScripts(RunningDeployment deployment)
+        protected void DeletePreferredScript(RunningDeployment deployment)
         {
-            var scripts = FindScripts(deployment);
+            var script = FindPreferredScript(deployment);
 
-            foreach (var script in scripts)
+            if (!string.IsNullOrEmpty(script))
             {
                 fileSystem.DeleteFile(script, FailureOptions.IgnoreFailure);
             }
         }
 
-        IEnumerable<string> FindScripts(RunningDeployment deployment)
+        string FindPreferredScript(RunningDeployment deployment)
         {
             var supportedScriptExtensions = scriptEngine.GetSupportedTypes();
             var searchPatterns = supportedScriptExtensions.Select(e => "*." + e.FileExtension()).ToArray();
-            return
-                from file in fileSystem.EnumerateFiles(deployment.CurrentDirectory, searchPatterns)
-                let nameWithoutExtension = Path.GetFileNameWithoutExtension(file)
-                where nameWithoutExtension.Equals(scriptFilePrefix, StringComparison.OrdinalIgnoreCase)
-                select file;
+
+            var files = from file in fileSystem.EnumerateFiles(deployment.CurrentDirectory, searchPatterns)
+                        let nameWithoutExtension = Path.GetFileNameWithoutExtension(file)
+                        let preferenceOrdinal = Array.IndexOf(supportedScriptExtensions, file.ToScriptType())
+                        where nameWithoutExtension.Equals(scriptFilePrefix, StringComparison.OrdinalIgnoreCase)
+                        orderby preferenceOrdinal
+                        select file;
+
+            var numFiles = files.Count();
+            var selectedFile = files.FirstOrDefault();
+
+            if (numFiles > 1)
+            {
+                var preferenceOrderDisplay = string.Join(" -> ", supportedScriptExtensions);
```

Comment:
Not sure an arrow is right here
```suggestion
                var preferenceOrderDisplay = string.Join(", ", supportedScriptExtensions);
```